shared/src/commonMain/kotlin/org/hyperskill/app/devices/data/source/DevicesCacheDataSource.kt

```diff
@@ -0,0 +1,9 @@
+package org.hyperskill.app.devices.data.source
+
+import org.hyperskill.app.devices.domain.model.Device
+
+interface DevicesCacheDataSource {
+    fun getCurrentDevice(): Result<Device>
```

Comment:
In that case `Result.failure` will be returned:
1. `Result.success` device returned successfully
2. `Result.failure` all other possible cases

Reply:
Why not nullable?
In my opinion, it's more clear.

Reply:
In case of `Result.failure` instead of nullable, the documentation is required, to make it clear.

Reply:
Think from another perspective - what does it mean `Result<Device?>`? It adds two levels of abstraction: an error and for some reason nullable value (why null value, what happen?).

In my opinion, it's more unclear rather than returning a non-nullable value always.

Reply:
> There are many similar usages in our codebase:

the same for a non-nullable

Reply:
Using `Result.failure` as a non-error result makes it impossible to understand, what is a real error, and what is a normal result of a function.

Reply:
As we have so different points of view, let's discuss it on PBR.

Reply:
OK